tests/appsec/test_appsec_utils.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+import pytest
+
+from ddtrace import constants
+import ddtrace.appsec.utils as utils
+from tests.utils import override_env
+
+
+@pytest.fixture(autouse=True)
+def reset_template_caches():
+    utils._HTML_BLOCKED_TEMPLATE_CACHE = None
+    utils._JSON_BLOCKED_TEMPLATE_CACHE = None
+    try:
+        yield
+    finally:
+        utils._HTML_BLOCKED_TEMPLATE_CACHE = None
+        utils._JSON_BLOCKED_TEMPLATE_CACHE = None
+
+
+def test_get_blocked_template_no_env_var_html():
+    with override_env(dict(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML="", DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON="")):
+        assert utils._get_blocked_template("text/html") == constants.APPSEC_BLOCKED_RESPONSE_HTML
+
+
+def test_get_blocked_template_no_env_var_json():
+    with override_env(dict(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML="", DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON="")):
+        assert utils._get_blocked_template("other") == constants.APPSEC_BLOCKED_RESPONSE_JSON
+        assert utils._get_blocked_template("application/json") == constants.APPSEC_BLOCKED_RESPONSE_JSON
+        assert utils._get_blocked_template("") == constants.APPSEC_BLOCKED_RESPONSE_JSON
+        assert utils._get_blocked_template(None) == constants.APPSEC_BLOCKED_RESPONSE_JSON
+
+
+def test_get_blocked_template_user_file_missing_html():
+    with override_env(dict(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML="missing.html")):
+        assert utils._get_blocked_template("text/html") == constants.APPSEC_BLOCKED_RESPONSE_HTML
+        assert utils._get_blocked_template("") == constants.APPSEC_BLOCKED_RESPONSE_JSON
+
+
+def test_get_blocked_template_user_file_missing_json():
+    with override_env(dict(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON="missing.json")):
+        assert utils._get_blocked_template("") == constants.APPSEC_BLOCKED_RESPONSE_JSON
```

Comment:
we should test with `application/json` as well

Reply:
`text/html` is what we use as the trigger in the code to determine html or json, but we should make sure the tests reflect real use cases rather than be written around implementation details of the code.

Reply:
True. Will be in next commit.